### PR TITLE
expat, bump version

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

File: dev-libs/expat/patches/expat-2.2.2.patchset [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+index c430541..69fe822 100644
+--- a/lib/xmlparse.c
++++ b/lib/xmlparse.c
+@@ -49,6 +49,11 @@
+ # include <bsd/stdlib.h>
+ #endif
+ 
++#ifdef __HAIKU__
++#if !defined(XML_POOR_ENTROPY)
++#define XML_POOR_ENTROPY 1
```
Comment (korli) [inline dev-libs/expat/patches/expat-2.2.2.patchset]: So until now, we had already a poor entropy, but weren't aware of it. Thus this PR doesn't change the situation (no regression). Is it right?

**Comment by Begasus** [comment]
At the moment the recipe works on all 3 platforms ...

**Comment by pulkomandy** [comment]
See the discussion in #1512, this is not going to work as Haiku really doesn't provide arc4random_buf. We should disable use of the function and be unsecure, until Haiku actually implements it.

**Comment by hartwork** [comment]
Expat 2.2.3 has just been released. It should work without `XML_POOR_ENTROPY` now and use `/dev/urandom`.

**Comment by Begasus** [comment]
Will check in a bit @hartwork , thanks!

**Comment by Begasus** [comment]
builds on all arches here ... 
only test fails on gcc2 ... 

Testing ...
gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o tests/runtests.o -c tests/runtests.c
/sources/expat-2.2.3/tests/runtests.c: In function `external_entity_dbl_handler':
/sources/expat-2.2.3/tests/runtests.c:8192: warning: `new_parser' might be used uninitialized in this function
gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o tests/chardata.o -c tests/chardata.c
gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o tests/minicheck.o -c tests/minicheck.c
/sources/expat-2.2.3/tests/minicheck.c: In function `srunner_run_all':
/sources/expat-2.2.3/tests/minicheck.c:138: warning: variable `tc' might be clobbered by `longjmp' or `vfork'
/sources/expat-2.2.3/tests/minicheck.c:143: warning: variable `i' might be clobbered by `longjmp' or `vfork'
gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o tests/memcheck.o -c tests/memcheck.c
/bin/sh ./libtool --verbose --mode=link gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -fno-strict-aliasing -o tests/runtests tests/runtests.o tests/chardata.o tests/minicheck.o tests/memcheck.o libexpat.la
libtool: link: gcc -I./lib -I. -g -O2 -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -fno-strict-aliasing -DHAVE_EXPAT_CONFIG_H -fno-strict-aliasing -o tests/.libs/runtests tests/runtests.o tests/chardata.o tests/minicheck.o tests/memcheck.o  ./.libs/libexpat.so -Wl,-rpath -Wl,/packages/expat-2.2.3-2/.self/lib
libtool: link: creating tests/runtests
g++ -I./lib -I. -g -O2 -fno-strict-aliasing  -DHAVE_EXPAT_CONFIG_H -o tests/runtestspp.o -c tests/runtestspp.cpp
In file included from /sources/expat-2.2.3/tests/runtestspp.cpp:6:
/sources/expat-2.2.3/tests/runtests.c:154: parse error before `__attribute__'
/sources/expat-2.2.3/tests/runtests.c: In function `void dummy_xdecl_handler(...)':
/sources/expat-2.2.3/tests/runtests.c:161: parse error before `__attribute__'
/sources/expat-2.2.3/tests/runtests.c:161: cannot declare static function inside another function
/sources/expat-2.2.3/tests/runtests.c: In function `void dummy_start_doctype_handler(...)':
/sources/expat-2.2.3/tests/runtests.c:161: confused by earlier errors, bailing out
Makefile:211: recipe for target 'tests/runtestspp.o' failed
make: *** [tests/runtestspp.o] Error 1
Warning: Command '['bash', '-c', '. /wrapper-script']' returned non-zero exit status 2
Error: Test has failed - stopping.

**Comment by Begasus** [comment]
PS, test on gcc5 is ok ...

**Comment by Begasus** [comment]
Updated the PR to the latest version, test is ok for gcc5
for gcc2 I still get an error on the test:


In file included from /sources/expat-2.2.3/tests/runtestspp.cpp:6:
/sources/expat-2.2.3/tests/runtests.c:154: parse error before `__attribute__'
/sources/expat-2.2.3/tests/runtests.c: In function `void dummy_xdecl_handler(...)':
/sources/expat-2.2.3/tests/runtests.c:161: parse error before `__attribute__'
/sources/expat-2.2.3/tests/runtests.c:161: cannot declare static function inside another function

File: dev-libs/expat/patches/expat-2.2.2.patchset (hunk 1 above)
Comment (Begasus) [inline dev-libs/expat/patches/expat-2.2.2.patchset]: The issue has been resolved in v2.2.3, but the poor entropy issue probably will be around for a while ...

**Comment by Begasus** [comment]
*need to change the revision*

**Comment by Begasus** [comment]
As the test fails on gcc2 I added the ? for x86_gcc2 (and also for arm as I can't test that one) ...

**Pull request merged**